projects/valtimo/zgw/src/lib/modules/documenten-api/components/documenten-api-upload-field-model/documenten-api-upload-field-modal.component.html

```diff
@@ -0,0 +1,115 @@
+  ~
+  ~ https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" basis,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<cds-modal
+  valtimoCdsModal
+  *ngIf="{
+    disabled: disabled$ | async,
+    languageItems: languageItems$ | async,
+    confidentialityLevelItems: confidentialityLevelItems$ | async,
+    statusItems: statusItems$ | async,
+    documentTypeItems: documentTypeItems$ | async,
+  } as obs"
+  [open]="type !== 'closed'"
+  size="sm"
+>
+  <cds-modal-header showCloseButton="true" (closeSelect)="onClose()">
+    <h3 cdsModalHeaderHeading>
+      {{
+        ('zgw.uploadFields.editModal.header' | translate) +
+          ' - ' +
+          ('zgw.uploadFields.keys.' + key | translate)
+      }}
+    </h3>
+  </cds-modal-header>
+
+  <section cdsModalContent>
+    <form [formGroup]="formGroup" class="uploadField-form">
+      @if (key === 'taal') {
+        <cds-combo-box
+          formControlName="selectedDefaultValue"
+          [label]="'zgw.uploadFields.fields.defaultValue' | translate"
+          [dropUp]="false"
+          (clear)="clear()"
+        >
+          <cds-dropdown-list [items]="obs.languageItems"></cds-dropdown-list>
+        </cds-combo-box>
+      } @else if (key === 'vertrouwelijkheidaanduiding') {
+        <cds-combo-box
+          formControlName="selectedDefaultValue"
+          [label]="'zgw.uploadFields.fields.defaultValue' | translate"
+          [dropUp]="false"
+          (clear)="clear()"
+        >
+          <cds-dropdown-list [items]="obs.confidentialityLevelItems"></cds-dropdown-list>
+        </cds-combo-box>
+      } @else if (key === 'status') {
+        <cds-combo-box
+          formControlName="selectedDefaultValue"
+          [label]="'zgw.uploadFields.fields.defaultValue' | translate"
+          [dropUp]="false"
+          (clear)="clear()"
+        >
+          <cds-dropdown-list [items]="obs.statusItems"></cds-dropdown-list>
+        </cds-combo-box>
+      } @else if (key === 'informatieobjecttype') {
+        <cds-combo-box
+          formControlName="selectedDefaultValue"
+          [label]="'zgw.uploadFields.fields.defaultValue' | translate"
+          [dropUp]="false"
+          (clear)="clear()"
+        >
+          <cds-dropdown-list [items]="obs.documentTypeItems"></cds-dropdown-list>
+        </cds-combo-box>
+      } @else if (
+        key === 'filename' ||
+        key === 'bestandsnaam' ||
+        key === 'titel' ||
+        key === 'beschrijving' ||
+        key === 'trefwoorden'
+      ) {
+        <cds-label>
+          {{ 'zgw.uploadFields.fields.defaultValue' | translate }}
+          <input cdsText formControlName="defaultValue" />
+        </cds-label>
+      }
```

Comment:
Use @switch

Reply:
Unfortunately, the `@Switch` cant do this nicely:
```
if (
        key === 'filename' ||
        key === 'bestandsnaam' ||
        key === 'titel' ||
        key === 'beschrijving' ||
        key === 'trefwoorden'
      )
```